client/src/pages/MedicationList.tsx

```diff
@@ -0,0 +1,173 @@
+import { Button } from '@/components/ui/button';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import {
+  Collapsible,
+  CollapsibleContent,
+  CollapsibleTrigger,
+} from '@/components/ui/collapsible';
+import {
+  BoxSelect,
+  Droplet,
+  HandCoins,
+  Pill,
+  Plus,
+  Syringe,
+  Tablets,
+  Wind,
+} from 'lucide-react';
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+type Medication = {
+  id: number;
+  name: string;
+  dosage: string;
+  form: string;
+  notes: string;
+  prescriber: string;
+  amount: number;
+  remaining: number;
+  userId: number;
+  createdAt: string;
+};
+
+const MedicationIcon = ({ type }) => {
+  switch (type) {
+    case 'Injection':
+      return <Syringe size={24} className="mr-2" />;
+    case 'Tablet':
+      return <Tablets size={24} className="mr-2" />;
+    case 'Topical':
+      return <HandCoins size={24} className="mr-2" />;
+    case 'Liquid':
+      return <Droplet size={24} className="mr-2" />;
+    case 'Patch':
+      return <BoxSelect size={24} className="mr-2" />;
+    case 'Inhaler':
+      return <Wind size={24} className="mr-2" />;
+    case 'Capsule':
+    default:
+      return <Pill size={24} className="mr-2" />;
+  }
+};
+
+export function MedicationList() {
+  const [medications, setMedications] = useState<Medication[]>([]);
+  const [openStates, setOpenStates] = useState<boolean[]>([]);
+  const [isAllExpanded, setIsAllExpanded] = useState(false);
+  const [error, setError] = useState<unknown>();
+
+  const userId = 1;
+  useEffect(() => {
+    const fetchMedications = async () => {
+      try {
+        const response = await fetch(`/api/medications/${userId}`);
+        if (!response.ok)
+          throw new Error(`Response status: ${response.status}`);
+        const medications = (await response.json()) as Medication[];
+        setMedications(medications);
+        setOpenStates(new Array(medications.length).fill(false));
+      } catch (error) {
+        setError(error);
+      }
+    };
+    fetchMedications();
+  }, []);
+
+  function toggleCard(index: number) {
+    setOpenStates((prevStates) => {
+      const newStates = [...prevStates];
+      newStates[index] = !newStates[index];
+      return newStates;
+    });
+  }
+
+  function toggleAll() {
+    setIsAllExpanded(!isAllExpanded);
+    setOpenStates(new Array(medications.length).fill(!isAllExpanded));
+  }
+
+  if (error) {
+    return (
+      <div>
+        Error Loading Medications:{' '}
+        {error instanceof Error ? error.message : 'Unknown Error'}
+      </div>
+    );
+  }
+  return (
+    <>
+      <section className="container pt-[110px] pb-[40px] ">
+        <div className="flex flex-wrap gap-2 items-center justify-center min-[400px]:justify-between button__group">
+          <Link to="/medications/add">
+            <Button size={'md'} variant={'secondary'}>
```

Comment:
Brackets are only needed when writing JavaScript expressions. If you are just assigning a string directly to the attribute, no brackets needed. This can be updated in multiple places